File: sources/extractor/internal/scip/testdata/fixtures/java-spring/src/main/java/example/CampaignService.java
package example;

/**
 * CampaignService is the business-logic layer between controller and
 * repository. Its softDelete method is the intermediate hop the
 * walker must follow.
 */
public class CampaignService {

    private final CampaignRepository repo;

    public CampaignService(CampaignRepository repo) {
        this.repo = repo;
    }

    public void softDelete(String id) {
        if (id == null || id.isEmpty()) {
            return;
        }
        Campaign c = repo.findById(id);
        if (c == null) {
            return;
        }
        repo.deleteById(id);
    }
}

File: sources/extractor/internal/scip/testdata/fixtures/kotlin-spring/src/main/kotlin/example/CampaignService.kt
package example

/**
 * CampaignService is the intermediate hop the walker must follow.
 */
class CampaignService(private val repo: CampaignRepository) {

    fun softDelete(id: String) {
        if (id.isEmpty()) return
        val campaign = repo.findById(id) ?: return
        repo.deleteById(campaign.id)
    }
}
